src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplay.cs

```diff
@@ -228,6 +228,14 @@ public static class SymbolDisplay
                 Debug.Assert(positionOpt < 0);
             }
 
+            // Do not leak unspeakable name of a Simple Program entry point through diagnostics,
+            // and, for consistency, with other display options, excluding TestFormat
+            if (format != (object)SymbolDisplayFormat.TestFormat &&
```

Comment:
>format != (object)SymbolDisplayFormat.TestFormat [](start = 16, length = 48)

We shouldn't test for specific format instances here. If it's important to change the behavior for some `SymbolDisplayFormat` settings, please add a `SymbolDisplayCompilerInternalOptions` value.

Reply:
> We shouldn't test for specific format instances here. If it's important to change the behavior for some SymbolDisplayFormat settings, please add a SymbolDisplayCompilerInternalOptions value.

I don't think I agree with this. The ```SymbolDisplayFormat.TestFormat``` is an internal instance that exists for the benefit of tests and this condition here is only for the benefit of tests. No external consumer can get to the format instance. It is not my intent to allow external consumers to configure display for this type and make a feature out of that.  

---
In reply to: [384803182](https://github.com/dotnet/roslyn/pull/41926#discussion_r384803182) [](ancestors = 384803182)